packages/block-editor/src/components/use-no-recursive-renders/index.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	createContext,
+	useCallback,
+	useContext,
+	useMemo,
+} from '@wordpress/element';
+
+const RenderedRefsContext = createContext( new Set() );
+
+// Immutably add to a Set
+function add( set, element ) {
+	const result = new Set( set );
+	result.add( element );
+	return result;
+}
+
+export default function useNoRecursiveRenders( uniqueId ) {
```

Comment:
JSDoc comment with an example should help create `README.md` once it gets promoted stable. In addition, it will make it easier to type the package with TS in the future.

Reply:
How's this? 1c3b036e2e

I'm not familiar with our latest typing practice. Do you have a specific React type you recommend for the returned provider? For now I just used `Function`.

Reply:
Yes, this is exactly what I meant 👍🏻 

If you think it deserves an example, you can use the following syntax:


https://github.com/WordPress/gutenberg/blob/1608a029b90a18938c2125e6a37579c4da95b1ef/packages/plugins/src/components/plugin-area/index.js#L38-L50